ci-operator/templates/openshift/installer/cluster-launch-installer-upi-e2e.yaml

```diff
@@ -627,6 +641,16 @@ objects:
           CONTROL_PLANE_1_IP="$(echo "${CONTROL_PLANE_IPS}" | cut -d, -f2)"
           CONTROL_PLANE_2_IP="$(echo "${CONTROL_PLANE_IPS}" | cut -d, -f3)"
 
+          # Stand up ami in public subnet with squid installed
+          # -- private subnet entries have route to public subnet
+          if [[ "${ENABLE_PROXY}" == "true" ]]; then
+
+            # call create-stack for cluster_master_nodes again?
```

Comment:
Currently the AWS UPI flow is:

1. [Fill in the `install-config.yaml`][1].
2. [Run `create manifests][2].
3. [Cross-platform `create ignition-configs`][3].
4. [Grab the infra-ID from `metadata.json`][4].
5. [Create the VPC, infra, and security stacks][5].

You're currently looking to create the proxy and plug its information in at this point, but that's pretty awkward, as you're wondering about in this comment.  It will take a bigger shuffle, but I think we want to pivot to:

1. Create the VPC, infra, and security stacks.
2. If `ENABLE_PROXY` is true, create the proxy in one of the public subnets.
3. Fill in the `install-config.yaml`, including the proxy settings.
4. Run `create manifests`.
5. Cross-platform `create ignition-configs`.

That means that you won't be able to use the installer-generated infra ID for at least the first three stacks, but that's not a big deal.  I'd suggest using something like:

```sh
INFRA_ID=$(echo "$(echo "${CLUSTER_NAME}" | cut -b -20)-$(uuidgen | sha256sum | cut -b -6)")
```

or some such.

[1]: https://github.com/openshift/release/blob/be9e978618f898e391986e18f771068a5a4942bd/ci-operator/templates/openshift/installer/cluster-launch-installer-upi-e2e.yaml#L395-L429
[2]: https://github.com/openshift/release/blob/be9e978618f898e391986e18f771068a5a4942bd/ci-operator/templates/openshift/installer/cluster-launch-installer-upi-e2e.yaml#L431
[3]: https://github.com/openshift/release/blob/be9e978618f898e391986e18f771068a5a4942bd/ci-operator/templates/openshift/installer/cluster-launch-installer-upi-e2e.yaml#L477
[4]: https://github.com/openshift/release/blob/be9e978618f898e391986e18f771068a5a4942bd/ci-operator/templates/openshift/installer/cluster-launch-installer-upi-e2e.yaml#L494
[5]: https://github.com/openshift/release/blob/be9e978618f898e391986e18f771068a5a4942bd/ci-operator/templates/openshift/installer/cluster-launch-installer-upi-e2e.yaml#L503-L568